packages/app/src/app/app.component.html

```diff
@@ -1 +1,26 @@
-<router-outlet />
+<!-- Main Application Container (gets blurred if the user is not authenticated
+with required roles) -->
+<div
+  [ngClass]="getAppContainerClasses()"
+  [attr.aria-hidden]="getMainContentAriaAttributes()['aria-hidden']"
+  [attr.inert]="getMainContentAriaAttributes()['inert'] || null"
+>
+  <!-- Main Application Content -->
+  <!-- This renders in the background even when splash is showing -->
+  <div class="main-content" *ngIf="shouldRenderMainContent()">
+    <!-- Router Outlet - Main App Routes -->
+    <main class="app-main" role="main">
+      <router-outlet></router-outlet>
+    </main>
+  </div>
+</div>
+
+<!-- Splash Screen Overlay (outside of blurred container) -->
+<!-- Only shows when user needs authentication or authorization -->
+<app-splash-screen
```

Comment:
Just FYI, could consider using CDK Overlay https://material.angular.dev/cdk/overlay/overview
but its API is a bit awkward and not sure it's necessarily better.